src/client_side.cc

```diff
@@ -4101,15 +4121,25 @@ ConnStateData::checkLogging()
 }
 
 bool
-ConnStateData::mayTunnelUnsupportedProto()
+ConnStateData::preserveDataForTunnellingUnsupportedProto()
 {
-    return Config.accessList.on_unsupported_protocol
 #if USE_OPENSSL
-           &&
-           ((port->flags.isIntercepted() && port->flags.tunnelSslBumping)
-            || (serverBump() && pinning.serverConnection))
+    // Tunneling may be possible when parsing the TLS client handshake
```

Comment:
If this method is not called when we are parsing the TLS client handshake, please remove that condition. I also suggest renaming this method to preserveHttpBytesForTunnellingUnsupportedProto() or similar, to underscore that it only deals with plain HTTP requests and decrypted HTTPS requests.

Reply:
Ok, this condition is removed. The preserveHttpBytesForTunnellingUnsupportedProto() name is used.
Resolved at commit e757c76